custom/icds_reports/views.py

```diff
@@ -1762,6 +1763,51 @@ def valid_state_names(self):
         return list(AwcLocation.objects.filter(aggregation_level=AggregationLevels.STATE, state_is_test=0).values_list('state_name', flat=True))
 
 
+@location_safe
+class DishaAPIViewV2(View):
+
+    def message(self, message_name):
+        state_names = ", ".join([state[0] for state in self.valid_states])
+        error_messages = {
+            "missing_date": "Please specify valid month and year",
+            "invalid_month": "Please specify a month that's older than a month and 5 days",
+            "invalid_state": "Please specify one of {} as state_name".format(state_names),
+        }
+        return {"message": error_messages[message_name]}
+
+    @method_decorator([api_auth, toggles.ICDS_DISHA_API_V2.required_decorator()])
+    def get(self, request, *args, **kwargs):
+        try:
+            month = int(request.GET.get('month'))
+            year = int(request.GET.get('year'))
+        except (ValueError, TypeError):
+            return JsonResponse(self.message('missing_date'), status=400)
+
+        query_month = date(year, month, 1)
+        today = date.today()
+        current_month = today - relativedelta(months=1) if today.day <= 5 else today
```

Comment:
Historically for DISHA API, we generated the data on the 5th day of each week in a celery task, but given this API is refreshed directly via API hit, this condition might not be necessary. But also fine to keep it, it just means the API can't be queried for a month during the first 5 days of that month.

Reply:
Also shouldn't the month always be the first day of the month (not any random day)?

Reply:
where does the 5th thing come from here?

Reply:
removint i